pennylane/operation.py

```diff
@@ -300,48 +300,149 @@ def hash(self):
+        This static method allows matrices to be computed
         directly without instantiating the operators first.
-
         To return the matrices of *instantiated* operators,
-        please use the :attr:`~.Operator.matrix` property instead.
+        please use the :attr:`~.Operator.matrix()` property instead.
+
+        Consults
+        If the matrix representation is not defined for this operator,
+        the method returns ``None``.
+
+        If ``wires`` and ``wire_order`` are provided, the
+        numerical representation considers the position of the
+        operator's wires in the wire order of the full vector space.
+        Otherwise, the defaults ``wires=[0,...,n]`` and ``wire_order=[0,...,n]``
+        are used, where ``n`` is the number of wires of this operator.
 
         **Example:**
 
-        >>> qml.RY._matrix(0.5)
-        >>> array([[ 0.96891242+0.j, -0.24740396+0.j],
-                   [ 0.24740396+0.j,  0.96891242+0.j]])
+        >>> qml.RY.compute_matrix(0.5)
+        array([[ 0.96891242, -0.24740396],
+               [ 0.24740396,  0.96891242]])
+
+        >>> qml.RY.compute_matrix(0.5, wires=["b"], wire_order=["a", "b"])
+        tensor([[ 0.96891242, -0.24740396, 0., 0.],
+                [ 0.24740396,  0.96891242, 0., 0.],
+                [ 0., 0., 0.96891242, -0.24740396],
+                [ 0., 0., 0.24740396,  0.96891242]
+                ])
+
+        >>> qml.RY.compute_matrix(0.5, wires=["b"], wire_order=["b", "a"])
+        tensor([[ 0.96891242, 0., -0.24740396, 0.],
+                [0.,  0.96891242, 0., -0.24740396],
+                [ 0.24740396, 0., 0.96891242, 0.],
+                [0.,  0.24740396, 0., 0.96891242+0.j],
+                ])
+
+        Args:
+            params (Iterable): trainable parameters that influence the matrix representation
+            wires (Iterable): wires of this operator
+            wire_order (Iterable): global wire order, must contain all wire labels in ``wires``
+            hyperparams (dict): other inputs that influence the matrix representation
 
         Returns:
-            array: matrix representation
+            tensor-like: matrix representation, or None if not defined
         """
-        raise NotImplementedError
+        base_matrix = cls._matrix(*params, **hyperparams)
+        interface = qml.math._multi_dispatch(params)  # pylint: disable=protected-access
 
-    @property
-    def matrix(self):
+        # representation is not defined
+        if base_matrix is None:
+            return None
+
+        # cannot have wire_order but no wires to interpret
+        if wires is None and wire_order is not None:
+            raise ValueError("Expected wires when wire order is defined, got None.")
+
+        # intersecting this case here because it is common
+        # and we want to avoid casting to Wires unless necessary
+        if wire_order is None:
+            # TODO: cast to interface?
```

Comment:
Shouldn't `_matrix` handle casting to the proper interface? `_matrix` would need to use the interface to construct it in a differentiable way, so it should already be converted?

Reply:
@albi3ro This is only the case for parametrized matrices, as the type can be inferred from the parameters. For non-parametrized matrices, casting will need to be done manually.

Reply:
Actually, I think this is responsible for my last failing tests in backprop devices' gradients...do you know where the casting is currently done, @josh146?